### PR TITLE
Ensure all test networks are within _with_ statements

### PR DESCRIPTION
**Pull request by cjen1-msft**
Looking at https://github.com/microsoft/CCF/actions/runs/18711838663/job/53362454045 it claims that the ledger directory already existed.
Afaict this can only happen if either:
- The previous delete of the workspace directory was non-atomic
- There existed a node which was writing to it and was not stopped successfully
- We have a bug in our startup behaviour

Of these the most likely is that some node was not properly killed at the end of a test, and one possible way would be if a network was started and not stopped.

In e2e_operations.py we have a couple of networks that get started within a test, and potentially are not closed.
(the one which is glaring is the run_recovery_unsealing_corrupt)

If we instead encapsulate all of these networks via the with keywork in python, then when they go out of scope they are always stopped (and there is logging of stacktraces etc).

This PR does so for all networks in e2e_operations.py
There are some other similar ones in - lts compatibility, but these seem less likely to be obviously correct.

I've also taken the opportunity to ensure that at the very least for these networks that they are not alising over the same directory.

### CODE REVIEW

File: tests/e2e_operations.py [hunk 1]
```diff
@@ -1067,7 +1067,9 @@ def run_empty_ledger_dir_check(args):
                 )
 
 
-def run_initial_uvm_descriptor_checks(args):
+def run_initial_uvm_descriptor_checks(const_args):
+    args = copy.deepcopy(const_args)
+    args.label += "_uvm_descriptor"
```
Comment (eddyashton) [inline tests/e2e_operations.py:1070]: I agree it'd be nicer to retain folders per-network (rather than overwriting), but I think manually tweaking the label here isn't ideal - it means actually reconstructing the path to find where your test output is becomes harder (`<ctest_name>_<runner_label>_<node_id|common>` vs `<ctest_name>_<runner_label>[_<maybe_a_test_suffix>]<node_id|common>`). Perhaps adding a `nested_label` to the `network(` constructor, and creating everything for this test nested within a dir of the existing label? Or splitting these out to top-level `ConcurrentRunner`s, so that matches the granularity of `network` objects?

Comment (cjen1-msft) [reply tests/e2e_operations.py:1070]: Having them as subdirectories of a test would probably work, we'd need to be careful in tests which have multiple networks that a nested_labelled network isn't run before a `nested_label=None` one as that would cause the non nested one to probably delete the nested one...

I'm inclined to do this adhoc for now, and then when we get around to fixing the test infra to have fixtures etc to think a bit about this.

Comment (achamayou) [reply tests/e2e_operations.py:1070]: Another way would be to add a method on the network contextmgr to created nested ones `with network.subnetwork()`, but it's clearly a "don't forget to do it" situation. Another possibility is to not catch the name from the labels, and/or to mix in the code path (module.method) where the network is created, to get something more unique.

**Review by eddyashton — APPROVED** [review]
